### ISSUE TITLE
Would be lovely to be able to send a Companion custom variable to a Dlive (or Avantis in my case) input fader rather than a preset level

### ISSUE THREAD

**Issue by adeattheback** [bitfocus/companion-module-allenheath-dlive-ilive#9]
It would make it easier to make a "mini mixer panel" on a Stream Deck page for those occasional users that only need access to a handful of channels. So, for each channel, a button to increase the value of the custom variable, and a second to decrease it and each press to send the value to the chosen fader. Thanks for listening!

**Comment by MeestorX** [comment]
I'm hoping someone else will take over this module as I don't have access to a dLive or iLive anymore. Hopefully they can add features like this and even more...

**Comment by Caleb-Wishart** [comment]
Can't promise anything but to get some more info on your request as I haven't really interacted with this module at all:

Is the idea to have control over a variable then send the value to a fader?

What kind of value would this need to be? A constant? A value from a drop-down? Etc?

Would it also be useful to have a variable to select a fader?
Such that you can select a fader and then a button that will send a value to whatever fader is currently "selected"?

**Comment by mattniebrugge** [comment]
I noticed a similar question in the Avantis module as well. From my perspective, it would be awesome to have a button/dial that adjusts a level by a certain, user-specified amount probably from in input box instead of a dropdown. This would be especially helpful when using a Streamdeck+, with the rotary dials, where fine tuning would be possible. 

I honestly think something almost exactly like they implemented in the ATEM module would be amazing for our use case: [](https://youtu.be/enNAXN9HnuM?t=76)

I think the idea of being able to select a fader could be interesting, but not sure it would be as helpful because the use case for Avantis is to not have to dig through layers to get to a channel. Now, if you were able to select the channel on the streamdeck bank instead, then I could see that being useful.

**Comment by SamSMithy15** [comment]
Yeah I agree. We recently got a Stream Deck Plus and wanted to try adjusting the faders through the dials, but it seems like it only sets them to a fixed amount. Seems like its been a good minute but still no updates.

**Comment by willosof** [comment]
If anyone has additional ideas or feature requests beyond what's already mentioned, now is the perfect time to share! We (Bitfocus) have hired @josephdadams to upgrade the module to v3 and enhance the Avantis module by implementing any missing features. Feedback, variables, and presets are already on our agenda, so don't hesitate to contribute your thoughts. 😊

Edit: I realise this is the dlive-module. Thought it was the Avantis :) But if there's lots of demand for changes to this module, we could look into this as well.

**Comment by adeattheback** [comment]
Thanks for this! Would be lovely for me to be able to have a single function that moves an output fader (channel, group, DCA, matrix, etc.) from a start value to an end value over a specified duration (e.g. -Inf -> -10db over 1 second). I know I can build this myself using relative delays already but it would be neat and presumably less processor intensive to have it all done in the background code.

**Comment by adeattheback** [comment]
It would be great if there were functions to access and control the AMM, even just toggling them on and off would be a help, but being able to toggle individual channels too and maybe select the AMM type would be icing on the cake.

**Comment by adeattheback** [comment]
I’d love to be able to select bank 1 – 6 on each screen…

**Comment by jswalden** [comment]
> Would be lovely for me to be able to have a single function that moves an output fader (channel, group, DCA, matrix, etc.) from a start value to an end value over a specified duration (e.g. -Inf -> -10db over 1 second).

The `allenheath-sq` module has [code](https://github.com/bitfocus/companion-module-allenheath-sq/blob/20f377048fab852ee8b564b753e78e1679e9c6ec/src/mixer/mixer.ts#L609) implementing a reasonably smooth fader move from a start level to an end level over a duration.  Anyone who wants it should feel free to copy it.  It's at least a minimum viable implementation of the feature.

I have some further idea of _recording_ when a fade is in progress, in order that a subsequent level-set that occurs during a fade can detect that fade, cancel it, and then do its own level-set.  As that code exists now, if you kicked off a 1-second fade, you could invoke a 0-second level set halfway into it that would work, but then would be overwritten by the last half of the initial fade -- which probably isn't desirable.  But it's low on the list of things to look at, because in practice IMO the big use case for that is where you _accidentally_ start a fade and then want to quickly cancel it -- and accidents are rarer than dealing with things that you actually want to have happen.
